R/extractFDAFeaturesMethods.R

```diff
@@ -177,20 +177,27 @@ extractFDAFPCA = function(pve = 0.99, npc = NULL) {
 #' @title Multiresolution feature extraction.
 #'
 #' @description
-#' The function extracts currently the mean of multiple segments of each curve and stacks them
-#' as features. The segments length are set in a hierachy way so the features
-#' cover different resolution levels.
+#' The function extracts the mean of multiple segments of each curve and extracts them
+#' as features. The segments length are set in a hierachy, so the features
+#' cover different resolution levels. This is done by sequentially dividing the
+#' functional up into smaller sub-curves of length l/2 and computing the mean of the
+#' sub-curves for a sliding window.
 #'
 #' @param res.level [\code{integer(1)}]\cr
-#'   The number of resolution hierachy, each length is divided by a factor of 2.
+#'   The resolution depth, each length is divided by a factor of 2.
 #' @param shift [\code{numeric(1)}]\cr
 #'   The overlapping proportion when slide the window for one step.
 #' @param curve.lens [\code{integer}]\cr
-#'   Curve subsequence lengths. Needs to sum up to the length of the functional.
+#'   Instead of splitting the curves in half from the top, a vector of sub-curves can be
+#'   can be specified. Specifies the curve subsequence's lengths.
+#'   Needs to sum up to the length of the functional.
 #' @return [\code{data.frame}].
 #' @export
 #' @family fda_featextractor
 extractFDAMultiResFeatures = function(res.level = 3L, shift = 0.5, curve.lens = NULL) {
+  assertInteger(res.level, len = 1L)
+  assertNumeric(shift, lower = 0L, upper = 1L, len = 1)
```

Comment:
again, there are special asserts for scalars in cm. please read the vignette to know those